base/uk.ac.stfc.isis.ibex.synoptic/src/uk/ac/stfc/isis/ibex/synoptic/xml/XMLUtil.java

```diff
@@ -33,87 +33,70 @@
 
 import org.xml.sax.SAXException;
 
-import uk.ac.stfc.isis.ibex.synoptic.model.desc.SynopticDescription;
+import com.google.common.base.Strings;
 
+/**
+ * Static utility class that deals with decoding/encoding XML into classes.
+ */
 public final class XMLUtil {
-	
-	private static final SchemaFactory SF = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
-    private static Schema schema; 
-
-    private static JAXBContext context;
-    private static Unmarshaller unmarshaller;
-    private static Marshaller marshaller;
-    
-    private XMLUtil() { }
     
+    private XMLUtil() {
+    }
+
 	/**
-     * Converts an input XML into a synoptic description.
+     * Converts an input XML into the given type.
      * 
-     * @param <T> the type to parse the XML into
-     * @param xml the synoptic XML received from the BlockServer
-     * @return the synoptic data converted into an instrument description
-     * @throws JAXBException XML Exception
-     * @throws SAXException XML Exception
+     * @param <T>
+     *            the type to parse the XML into
+     * @param xml
+     *            the XML to convert
+     * @param clazz
+     *            the type to parse the XML into
+     * @return the xml data converted into the specified type
+     * @throws JAXBException
+     *             XML Exception thrown if the conversion failed
      */
-    public static synchronized <T> T fromXml(String xml) throws JAXBException, SAXException {
-		if (context == null) {
-			initialise();
-		}
-	         
+    @SuppressWarnings("unchecked")
+    public static synchronized <T> T fromXml(String xml, Class<T> clazz) throws JAXBException {
```

Comment:
http://stackoverflow.com/questions/2529974/why-do-java-programmers-like-to-name-a-variable-clazz